enhancements/sig-architecture/70-spread-policy/README.md

```diff
@@ -0,0 +1,301 @@
+
+**Example.** We want four clusters from aws to put our backend service. Further, we want an even spread across regions and an affinity spread across zones. Specifically, for the region us-east-1, we want to first put the workload to us-east-1a then us-east-1b. Assuming all zones here can provide enough clusters. Then, the scheduling result should contain two clusters from us-east-1a and other two clusters from us-west-1a.
+
+### Use Case 6: Spread with Ratio (Out of Scope)
+
+Sometimes, we may need to spread the workload with a specified ratio. Let us consider a scenario where most of the access to the workload is from the east coast of the US, even spread between us-east-1 and us-west-1 would be inappropriate. Instead, we want to place more workload on us-east-1 than us-west-1. Besides, we want to control such preferences by user-specified configurations.
+
+**Example.** We want six clusters to deploy our backend service. Moreover, we want 2/3 of them to be located in us-east-1 and 1/3 to be located in us-west-1 since there are more clients from the east coast of the US.
+
+## Goal
+Generally, we have two goals for the spread policy:
+1. Cover the full functionality in the Motivation section;
+2. Implement a high-performance and scalable mechanism that works for thousands of clusters efficiently.
+
+## Proposal
+
+### API Spec
+```go
+type PlacementSpec struct {
+	// ...
+
+	// SpreadConstraints defines how placement decision should be distributed among a
+	// set of clusters.
+	SpreadConstraints []SpreadConstraintsTerm `json:"spreadConstraints,omitempty"`
+}
+
+// SpreadConstraintsTerm defines a terminology to spread placement decisions
+type SpreadConstraintsTerm struct {
+	// Type is the type of SpreadConstraintsTerm. It could be Even or Affinity.
+	// +required
+	Type string `json:"type"`
+
+	// TopologyKey is either a label key or a cluster claim name of ManagedClusters
+	// +required
+	TopologyKey string `json:"topologyKey"`
+
+	// TopologyKeyType indicates the type of TopologyKey. It could be Label or Claim.
+	// +required
+	TopologyKeyType string `json:"topologyKeyType"`
+
+	// TopologyWeights indicates the optional weights of topologies, valid when Type
+	// is Affinity.
+	TopologyWeights []TopologyWeightTerm `json:"topologyWeights,omitempty"`
+
+	// MaxSkew describes the degree to which the workload may be unevenly distributed,
+	// valid when Type is Even. If omitted, the scheduler treat the even constraint a soft
+	// constraint.
+	MaxSkew *int32 `json:"maxSkew,omitempty"`
+
+	// Order indicates the order of even constraints. It is valid when Type is Even and
+	// required only when multiple even constraints are used. To spread the workload,
+	// the scheduler takes the even constraints with a smaller order value into prior consideration.
+	Order int32 `json:"order,omitempty"`
```

Comment:
order seems strange. If we finally calculate this as a prioritizer, should this be weight?

Reply:
I have changed this field to `weight`. But the value of this field is designed to represent the hierarchical structure of topology keys, and is not used directly for calculating scores.

For example, we have three topology keys: `provider`, `region` and `zone`. Their weight are a, b, and c, respectively. Then any combination of three numbers that ensures a > b > c is considered appropriate.

Reply:
yeah, your description indicates that is could either be weight or priority.